runsc/boot/gofer_conf.go

```diff
@@ -184,7 +191,7 @@ func (g GoferMountConf) ShouldUseErofs() bool {
 
 // valid returns true if this is a valid gofer mount config.
 func (g GoferMountConf) valid() bool {
-	return g.Lower < LowerMax && g.Upper < UpperMax && (g.Lower != NoneLower || g.Upper != NoOverlay)
+	return g.Lower < LowerMax && g.Upper < UpperMax && (g.Lower != NoneLower || g.Upper != NoOverlay) && (g.UpperSize == "" || g.ShouldUseOverlayfs())
```

Comment:
I don't understand this change. When ShouldUseOverlayfs() returns false, it can be that we want to apply a tmpfs-only. In that case, why do we want size==""?

I think size does not apply when upper == NoOverlay, since here is nothing to apply the size on. Upper can be NoOverlay when the rootfs is being mounted readonly (in the OCI spec, Spec.Root.Readonly controls this). In such situations, we don't want to error out. Instead we can just ignore the size when upper is not present.

Revert this line.